extensions/json-jackson/src/main/java/io/deephaven/json/jackson/IntMixin.java

```diff
@@ -0,0 +1,154 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.json.jackson;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import io.deephaven.base.ArrayUtil;
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.json.IntOptions;
+import io.deephaven.json.jackson.IntValueProcessor.ToInt;
+import io.deephaven.qst.type.Type;
+import io.deephaven.util.QueryConstants;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static io.deephaven.util.type.ArrayTypeUtils.EMPTY_INT_ARRAY;
+
+final class IntMixin extends Mixin<IntOptions> implements ToInt {
+
+    public IntMixin(IntOptions options, JsonFactory factory) {
+        super(factory, options);
+    }
+
+    @Override
+    public int numColumns() {
+        return 1;
+    }
+
+    @Override
+    public Stream<List<String>> paths() {
+        return Stream.of(List.of());
+    }
+
+    @Override
+    public Stream<Type<?>> outputTypesImpl() {
+        return Stream.of(Type.intType());
+    }
+
+    @Override
+    public ValueProcessor processor(String context, List<WritableChunk<?>> out) {
+        return new IntValueProcessor(out.get(0).asWritableIntChunk(), this);
+    }
+
+    @Override
+    public int parseValue(JsonParser parser) throws IOException {
+        switch (parser.currentToken()) {
+            case VALUE_NUMBER_INT:
+                return parseFromInt(parser);
+            case VALUE_NUMBER_FLOAT:
+                return parseFromDecimal(parser);
+            case VALUE_STRING:
+            case FIELD_NAME:
+                return parseFromString(parser);
+            case VALUE_NULL:
+                return parseFromNull(parser);
+        }
+        throw Parsing.mismatch(parser, int.class);
+    }
+
+    @Override
+    public int parseMissing(JsonParser parser) throws IOException {
+        return parseFromMissing(parser);
+    }
+
+    @Override
+    RepeaterProcessor repeaterProcessor(boolean allowMissing, boolean allowNull, List<WritableChunk<?>> out) {
+        return new IntRepeaterImpl(out.get(0).asWritableObjectChunk()::add, allowMissing, allowNull);
+    }
+
+    final class IntRepeaterImpl extends RepeaterProcessorBase<int[]> {
+
+        public IntRepeaterImpl(Consumer<? super int[]> consumer, boolean allowMissing, boolean allowNull) {
+            super(consumer, allowMissing, allowNull, null, null);
```

Comment:
Why aren't we taking `onMissing` or `onNull`, if we're being asked to care about `allowMissing` or `allowNull`?

Reply:
Theoretically, I agree. We should want to allow users to specify null, empty array, or a specific array, for each case. It's possible to plumb a generic array types throughout the impl

```java
abstract RepeaterProcessor repeaterProcessor(boolean allowMissing, boolean allowNull, AT onMissing, AT onNull)
```

, and similarly on the Value side, but it's a lot of work right now for a little bit of value.